packages/@aws-cdk/aws-apigatewayv2/lib/http/integration.ts

```diff
@@ -191,11 +192,46 @@ export interface HttpRouteIntegrationBindOptions {
 /**
  * The interface that various route integration classes will inherit.
  */
-export interface IHttpRouteIntegration {
+export abstract class HttpRouteIntegration {
+  private integration?: HttpIntegration;
+
+  /**
+   * Internal method called when binding this integration to the route.
+   * @internal
+   */
+  public _bindToRoute(options: HttpRouteIntegrationBindOptions): { readonly integrationId: string } {
+    if (this.integration && this.integration.httpApi.node.addr !== options.route.httpApi.node.addr) {
+      throw new Error('A single integration cannot be associated with multiple APIs.');
+    }
+
+    if (!this.integration) {
+      const config = this.bind(options);
+
+      this.integration = new HttpIntegration(options.scope, `HttpIntegration-${hash(config)}`, {
+        httpApi: options.route.httpApi,
+        integrationType: config.type,
+        integrationUri: config.uri,
+        method: config.method,
+        connectionId: config.connectionId,
+        connectionType: config.connectionType,
+        payloadFormatVersion: config.payloadFormatVersion,
+        secureServerName: config.secureServerName,
+        parameterMapping: config.parameterMapping,
+      });
+
+      function hash(x: any) {
+        const stringifiedConfig = JSON.stringify(Stack.of(options.scope).resolve(x));
```

Comment:
I don't know if it can happen with `Stack`s (maybe `resolve()` takes care of that), but if the order of the fields change or if the presence or absence of whitespaces change, the hash will be different.

Reply:
Whitespace is not a problem here. Ordering maybe. I'll check.

Reply:
Ordering is actually a bigger problem here than I thought. 

This PR is effectively not changing the logic of hashing. Just moving it to a different place (previously `integration-cache.ts`).

Do you mind if I take this on as a follow up PR? Easier to review as well.